cluster/ci/monitoring/prometheus_rule_prow.yaml

```diff
@@ -205,3 +205,43 @@ spec:
       for: 1m
       labels:
         severity: slack
+  - name: ghproxy
+    rules:
+    - alert: ghproxy-status-code-abnormal-3XX
+      annotations:
+        message: ghproxy has {{ $value | humanize }}% of status code 3XX for over
+          1 minute.
+      expr: |
+        sum(github_request_duration_count{status=~"3..", status!="304|404"})/sum(github_request_duration_count{status!="404"}) * 100 > 5
+      for: 1m
+      labels:
+        severity: slack
+    - alert: ghproxy-status-code-abnormal-4XX
+      annotations:
+        message: ghproxy has {{ $value | humanize }}% of status code 4XX for over
+          1 minute.
+      expr: |
+        sum(github_request_duration_count{status=~"4..", status!="304|404"})/sum(github_request_duration_count{status!="404"}) * 100 > 5
+      for: 1m
+      labels:
+        severity: slack
+    - alert: ghproxy-status-code-abnormal-5XX
+      annotations:
+        message: ghproxy has {{ $value | humanize }}% of status code 5XX for over
+          1 minute.
+      expr: |
+        sum(github_request_duration_count{status=~"5..", status!="304|404"})/sum(github_request_duration_count{status!="404"}) * 100 > 5
+      for: 1m
+      labels:
+        severity: slack
+    - alert: ghproxy-running-out-github-tokens-in-a-hour
+      annotations:
+        message: ghproxy is running out of github tokens {{ $labels.token_hash }}
+          in a hour.
+      expr: |
+        github_token_usage{job="ghproxy"} <  1500
+        and
+        predict_linear(github_token_usage{job="ghproxy"}[1h], 1 * 3600) < 0
```

Comment:
I guess we can start with this but look at the graph -- you can't do a linear prediction across a reset. We want to do a linear prediction only for the last period, which we could figure out the timing for by looking at the reset time for the same hash

Reply:
I see your point.

crazy talking:
Would it be a problem for any usage of `predict_linear`, for example, [disk free space](https://www.robustperception.io/reduce-noise-from-disk-space-alerts#more-614). What if a cron job running every hour to clean up a huge trunk of data. Would it make the prediction useless?

the logic to fix in a specific linear ... the problem is how we do it by prometheus query? :)

Reply:
Yeah, hard to say. Since we know exactly when the resets are based on the other metric we can likely plumb it through. We should ask the monitoring team and/or chance. This can merge as is as long as we don't get spammed.

Reply:
Yeah.
Asked monitoring team via [slack](https://coreos.slack.com/archives/C0VMT03S5/p1565901948456800).